Platon/backend/app/activity_stream/views.py

```diff
@@ -0,0 +1,212 @@
+    "summary": fields.String,
+    "type": fields.String,
+    "actor": fields.Nested(activity_stream_item_actor_model),
+    "object": fields.Nested(activity_stream_item_object_model),
+    "target": fields.Nested(activity_stream_item_target_model)
+})
+
+activity_stream_model = api.model('Activity Stream', {
+    "@context": fields.String,
+    "summary": fields.String,
+    "type": fields.String,
+    "id": fields.Integer,
+    "orderedItems": fields.List(
+        fields.Nested(activity_stream_item_model)
+    )
+})
+
+@activity_stream_ns.route("")
+class ActivityStreamAPI(Resource):
+    
+    @api.doc(responses={401 : 'Account Problems', 400 : 'Input Format Error' ,500 : ' Database Connection Error', 404: 'Not found'})
+    @api.response(200, 'Success', activity_stream_model)
+    @api.expect(get_activity_stream_parser)
+    @login_required
+    def get(user_id, self):
+        '''
+            Returns activity stream of an user.
+        '''
+        '''
+            # Pagination functionality
+            number_of_pages = 1
+            if form.page.data is not None and form.per_page.data is not None:
+                per_page = form.per_page.data
+                number_of_pages = math.ceil(len(return_list) / per_page)
+                # Assign the page index to the maximum if it exceeds the max index
+                page = form.page.data if form.page.data < number_of_pages else number_of_pages-1
+                return_list = return_list[page*per_page:(page+1)*per_page]
+            return make_response(jsonify({'number_of_pages': number_of_pages ,'result': return_list}),200)
+        '''
+
+        # Parses the form data.
+        form = GetActivityStreamForm(request.args)
+        
+        if form.validate():
+
+            # Get current user's followings list.
+            try:
+                followings_list = Follow.query.filter(Follow.follower_id==user_id).all()
+            except:
+                return make_response(jsonify({"error" : "The server is not connected to the database."}), 500)
+
+            followings_ids_list = []
+            for followings_item in followings_list:
+                followings_ids_list.append(followings_item.following_id)
+
+            # Get current user's workspaces list.
+            try:
+                contributions = Contribution.query.filter(Contribution.user_id==user_id).all()
+            except:
+                return make_response(jsonify({"error": "Database Connection Error"}),500)
+            
+            workspace_ids_list = []
+            for contribution in contributions:
+                if contribution.is_active == 0:
+                    continue
+                try:
+                    ws = Workspace.query.get(contribution.workspace_id)
+                except:
+                    return make_response(jsonify({"error": "Database Connection Error"}),500)
+
+                workspace_ids_list.append(ws.id)
```

Comment:
How and where is this list used? Could you please explain? :sweat_smile:  Maybe I missed the part it is used.

Reply:
My bad. Activity Stream is incomplete. Currently, it only checks the activities of the following users. However, it should also show the activities of user's workspaces. That's why I've stored all the workspaces that user is active. I'll use this information later. Thanks